### PR TITLE
issue 754 : added 'update' method to the Session class

### PR DESCRIPTION
**Pull request by alixdamman**
needs changelog

### CODE REVIEW

File: larray/core/session.py [hunk 1]
```diff
@@ -132,6 +132,103 @@ def add(self, *args, **kwargs):
...
+        arr2: z (3) [int32]
+        z: z ['z0' 'z1' 'z2'] (3)
+        arr3: x, z (3 x 3) [int32]
+        """
+        if other is None:
+            pass
+        elif hasattr(other, 'items'):
+            for k, v in other.items():
+                self[k] = v
+        elif hasattr(other, '__getitem__'):
```
Comment (gdementen) [inline larray/core/session.py]: seems like a strange way to test other is an iterable. Is there a reason not to use isinstance(other, Iterable) ?

Comment (alixdamman) [reply larray/core/session.py]: No, there is no good reason. I'll change that.

File: larray/core/session.py [hunk 2]
```diff
@@ -132,6 +132,103 @@ def add(self, *args, **kwargs):
...
+
+        Parameters
+        ----------
+        other: Session or dict-like object or iterable with key/value pairs
+            Object containing key/value pairs to add or modify.
+        **kwargs:
+            If keyword arguments are specified, the session is then updated with those key/value pairs
+            (e.g.: ses.update(pop=pop, births=births, deaths=deaths)).
+
+        Returns
```
[the remark this answers is not in the document]
Comment (alixdamman) [reply larray/core/session.py]: Seems you're right. Numpy doesn't specify a `Returns` section if the operation is made inplace ( https://docs.scipy.org/doc/numpy-1.12.0/reference/generated/numpy.ufunc.at.html )

**Comment by gdementen** [comment]
LGTM, modulo changelog

**Pull request merged**